core/src/main/java/com/scalar/db/transaction/consensuscommit/CrudHandler.java

```diff
@@ -73,6 +73,20 @@ private Optional<Result> createGetResult(Snapshot.Key key, List<String> projecti
   }
 
   public List<Result> scan(Scan scan) throws CrudException {
+    List<Result> results = scanInternal(scan);
+
+    // We verify if this scan does not overlap previous writes after the actual scan. For a
+    // relational scan, this must be done here, using the obtained results. This is because the
```

Comment:
> Or you meant scanSet is updated by scanInternal and is used in the check logic (isWriteSetOverlappedWithRelational)?

Right, we use the obtained results put in the `scanSet` by `scanInternal`.

Reply:
"using the obtained results" was a bit confusing since the check only uses keys in the scan set (not returned results above). I changed the comment a little in 19f4ada.